From ca4c769ebfd7aa545ace9ebe8df7e250619c1c8e Mon Sep 17 00:00:00 2001
From: Arnaldo Garcia Rincon <agarrcia@amazon.com>
Date: Wed, 12 Nov 2025 03:12:41 +0000
Subject: [PATCH] config/efa.cmake: execute config tests in serial

Signed-off-by: Arnaldo Garcia Rincon <agarrcia@amazon.com>
---
 config/efa.cmake | 10 ++--------
 1 file changed, 2 insertions(+), 8 deletions(-)

diff --git a/config/efa.cmake b/config/efa.cmake
index 5bdad0b..02fc5f4 100644
--- a/config/efa.cmake
+++ b/config/efa.cmake
@@ -53,15 +53,10 @@ include(ProcessorCount)
 ProcessorCount(max_process)
 
 function(try_compile prologue body success_def fail_def)
-  # Rate limit processes
-  process_rl()
   set_conf_tmp_dir("${prologue}" "${body}")
-  execute_process(COMMAND ${CMAKE_SOURCE_DIR}/config/runbg.sh ${CMAKE_SOURCE_DIR}/config/compile_conftest.sh ${CMAKE_CURRENT_BINARY_DIR}/${tmp_dir} ${KERNEL_DIR} "${success_def}" "${fail_def}"
+  execute_process(COMMAND ${CMAKE_SOURCE_DIR}/config/compile_conftest.sh ${CMAKE_CURRENT_BINARY_DIR}/${tmp_dir} ${KERNEL_DIR} "${success_def}" "${fail_def}"
     WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
-    OUTPUT_STRIP_TRAILING_WHITESPACE
-    OUTPUT_VARIABLE pid)
-  list(APPEND pids ${pid})
-  set(pids "${pids}" CACHE INTERNAL "")
+    RESULT_VARIABLE res)
 endfunction()
 
 function(try_compile_dev_or_ops fp_name prologue fn success_def fail_def)
@@ -535,5 +530,4 @@ struct ib_mr *efa_reg_user_mr_dmabuf(struct ib_pd *ibpd, u64 start, u64 length,
   "
   HAVE_REG_USER_MR_DMAH "")
 
-wait_for_pids()
 message("-- Inspecting kernel - done")
-- 
2.50.1

